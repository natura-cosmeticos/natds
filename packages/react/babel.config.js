const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
