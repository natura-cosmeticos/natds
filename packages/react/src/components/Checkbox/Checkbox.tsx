import React from 'react'
import { CheckboxProps } from './Checkbox.props'
import Ripple from '../Ripple'
import styles from './Checkbox.styles'

const Checkbox = ({
  checked = false,
  disabled = false,
  id,
  indeterminate = false,
  onChange,
  testID = `ds-checkbox-${id}`,
  value
}: CheckboxProps): JSX.Element => {
  const { checkbox, wrapper } = styles({ indeterminate })

  const getRippleColor = checked ? 'primary' : 'highlight'

  return (
    <Ripple
      isCentered
      hideOverflow={false}
      disabled={disabled}
      color={getRippleColor}
      animationDuration={500}
      showHover
      focus
    >
      <div className={wrapper}>
        <input
          data-testid={testID}
          checked={checked}
          className={checkbox}
          disabled={disabled}
          id={id}
          onChange={onChange}
          type="checkbox"
          value={value}
        />
        <label htmlFor={id} />
      </div>
    </Ripple>
  )
}

export default Checkbox
