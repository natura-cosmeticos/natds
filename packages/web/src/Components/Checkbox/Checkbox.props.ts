import { CheckboxProps } from '@material-ui/core'

export type CheckboxColor = 'primary' | 'secondary' | 'default';
export type CheckboxEdge = CheckboxProps['edge'];
export type ICheckboxProps = Omit<CheckboxProps, 'translate'>
