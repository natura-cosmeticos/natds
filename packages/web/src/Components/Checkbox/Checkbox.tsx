import * as React from 'react'
import MaterialCheckbox, { CheckboxProps as ICheckboxProps } from '@material-ui/core/Checkbox'

export { ICheckboxProps } from './Checkbox.props'

/**
 * For more advanced usages, check [Material UI Checkbox docs](https://material-ui.com/components/checkboxes/)
 *
 * ## Importing
 *
 * ```
 * import { Checkbox } from '@naturacosmeticos/natds-web';
 * ```
 */
export const Checkbox = React.forwardRef<HTMLButtonElement, ICheckboxProps>((
  props: ICheckboxProps, ref
) => {
  const { indeterminateIcon, ...otherProps } = props

  /**
   * @todo adopt default indeterminate icon from `natds-icons`
   */
  // const defaultIndeterminateIcon = <Icon name={"filled-toggle-indeterminatecheckbox"} />;

  return (
    <MaterialCheckbox
      indeterminateIcon={indeterminateIcon}
      ref={ref}
      {...otherProps}
    />
  )
})

Checkbox.displayName = 'Checkbox'

export default Checkbox
