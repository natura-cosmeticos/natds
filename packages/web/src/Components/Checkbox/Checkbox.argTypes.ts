import { ArgType } from '@storybook/addons'
import { ICheckboxProps } from './Checkbox.props'
import { colors } from './__fixtures__/colors'
import { booleanArgType } from '../../../.storybook/argTypes/booleanArgType'
import { objectArgType } from '../../../.storybook/argTypes/objectArgType'

const argTypes : Record<keyof ICheckboxProps | string, ArgType> = {
  checked: {
    ...booleanArgType,
    description: 'If `true`, the component is checked.'
  },
  checkedIcon: {
    description: 'The icon to display when the component is checked.',
    table: { type: { summary: 'node' } }
  },
  classes: {
    ...objectArgType,
    description: 'Override or extend the styles applied to the component. See [CSS API](https://material-ui.com/pt/api/switch/#css) for more details.'
  },
  color: {
    control: { options: colors, type: 'inline-radio' },
    description: 'The color of the component. It supports those theme colors that make sense for this component.',
    table: {
      defaultValue: { summary: '"secondary"' },
      type: { summary: '"primary" | "secondary" | "default"' }
    }
  },
  disabled: {
    ...booleanArgType,
    description: 'If `true`, the checkbox will be disabled.'
  },
  disableRipple: {
    ...booleanArgType,
    description: 'If `true`, the ripple effect will be disabled.'
  },
  icon: {
    description: 'The icon to display when the component is unchecked.',
    table: { type: { summary: 'node' } }
  },
  id: {
    control: { type: 'text' },
    description: 'The id of the input element.',
    table: { type: { summary: 'string' } },
    type: { name: 'string', required: false }
  },
  indeterminate: {
    ...booleanArgType,
    description: `
If \`true\`, the component appears indeterminate.

This does not set the native input element to indeterminate due to inconsistent behavior across browsers.
However, we set a \`data-indeterminate\` attribute on the input.
`,
    table: {
      ...booleanArgType.table,
      defaultValue: { summary: false }
    }
  },
  indeterminateIcon: {
    description: 'The icon to display when the component is indeterminate.',
    table: {
      defaultValue: { summary: '<IndeterminateCheckBoxIcon />' },
      type: { summary: 'node' }
    }
  },
  inputProps: {
    ...objectArgType,
    description: '[Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the input element.'
  },
  inputRef: {
    description: 'Pass a `ref` to the `input` element.',
    table: { type: { summary: 'ref' } }
  },
  onChange: {
    description: 'Callback fired when state changes',
    table: {
      type: { summary: 'func', detail: 'function(event: object) => void' }
    }
  },
  required: {
    ...booleanArgType,
    description: 'If `true`, the `input` element will be required.'
  },
  type: {
    control: { type: 'text' },
    description: 'The `input` component prop type.',
    table: {
      type: { summary: 'string' }
    },
    type: { name: 'string' }
  },
  value: {
    description: 'The value of the component. The DOM API casts this to a `string`.',
    table: {
      type: { summary: 'any' }
    }
  }
}

export default argTypes
