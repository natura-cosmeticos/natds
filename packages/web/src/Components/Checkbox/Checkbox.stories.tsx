import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import Checkbox, { ICheckboxProps } from './Checkbox'
import { colors } from './__fixtures__/colors'
import argTypes from './Checkbox.argTypes'
import edges from './__fixtures__/edges'

export default {
  argTypes,
  component: Checkbox,
  title: 'Components/Checkbox (selection control)'
} as Meta

const Template : Story<ICheckboxProps> = (args: ICheckboxProps) => {
  const [checkedState, setChecked] = React.useState<ICheckboxProps['checked']>(args.checked)

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>, checked: ICheckboxProps['checked']) => {
    setChecked(checked)
    args.onChange(event, checked)
  }

  return <Checkbox {...args} checked={checkedState} onChange={handleChange} />
}

const TemplateWithCheckedAndUnchecked : Story<ICheckboxProps> = (args: ICheckboxProps) => (
  <>
    <Checkbox {...args} checked={false} />
    <Checkbox {...args} checked />
  </>
)

export const Playground : Story<ICheckboxProps> = Template.bind({})
Playground.args = {
  checked: true,
  color: colors.secondary,
  disabled: false,
  indeterminate: false
}

export const PrimaryColor : Story<ICheckboxProps> = TemplateWithCheckedAndUnchecked.bind({})
PrimaryColor.args = {
  color: colors.primary
}

export const SecondaryColor : Story<ICheckboxProps> = TemplateWithCheckedAndUnchecked.bind({})
SecondaryColor.args = {
  color: colors.secondary
}

export const DefaultColor : Story<ICheckboxProps> = TemplateWithCheckedAndUnchecked.bind({})
DefaultColor.args = {
  color: colors.default
}

export const Enabled : Story<ICheckboxProps> = TemplateWithCheckedAndUnchecked.bind({})
Enabled.args = {
  disabled: false
}

export const Disabled : Story<ICheckboxProps> = TemplateWithCheckedAndUnchecked.bind({})
Disabled.args = {
  disabled: true
}

export const Indeterminate : Story<ICheckboxProps> = TemplateWithCheckedAndUnchecked.bind({})
Indeterminate.args = {
  indeterminate: true
}

export const ForTableHeader : Story<ICheckboxProps> = Template.bind({})
ForTableHeader.args = {
  color: colors.primary,
  inputProps: {
    'aria-label': 'Select all items'
  }
}

export const ForTableCell : Story<ICheckboxProps> = Template.bind({})
ForTableCell.args = {
  color: colors.primary,
  inputProps: {
    'aria-label': 'Select item'
  }
}

export const ForListItemSecondaryAction : Story<ICheckboxProps> = Template.bind({})
ForListItemSecondaryAction.args = {
  checked: true,
  edge: edges.end
}
