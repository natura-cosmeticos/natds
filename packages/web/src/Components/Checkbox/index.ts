import Checkbox from './Checkbox'

export { ICheckboxProps } from './Checkbox.props'
export { Checkbox }
export default Checkbox
