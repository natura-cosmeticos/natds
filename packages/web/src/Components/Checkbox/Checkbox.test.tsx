import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import Checkbox from './Checkbox'

describe('Checkbox component', () => {
  describe('when no props are provided', () => {
    it('should match snapshot', () => {
      const testRenderer = TestRenderer.create(<Checkbox />)

      expect(testRenderer).toMatchSnapshot()
    })
  })
  it('when color is primary', () => {
    const testRenderer = TestRenderer.create(<Checkbox color="primary" />)

    expect(testRenderer).toMatchSnapshot()
  })
  it('when checked', () => {
    const testRenderer = TestRenderer.create(<Checkbox checked />)

    expect(testRenderer).toMatchSnapshot()
  })
  it('when disabled', () => {
    const testRenderer = TestRenderer.create(<Checkbox disabled />)

    expect(testRenderer).toMatchSnapshot()
  })
})
