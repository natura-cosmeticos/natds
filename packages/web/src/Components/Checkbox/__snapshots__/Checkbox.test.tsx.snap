// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component when checked 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      checked={true}
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </span>
</span>
`;

exports[`Checkbox component when color is primary 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</span>
`;

exports[`Checkbox component when disabled 1`] = `
<span
  aria-disabled={true}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-disabled-3 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
</span>
`;

exports[`Checkbox component when no props are provided should match snapshot 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      className="PrivateSwitchBase-input-4"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
</span>
`;
